### PR TITLE
Doc/example update

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Updates the documentation examples.  Partially addresses #528 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

Updating my pysat system and running the example.  NOTE that the new and original output figures differ, and I couldn't figure out the original intent or the `B_flag` output from the comments.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by aburrell** [comment]
@rstoneback this is your example, so up to you to finish fixing where I have failed!

Also, I had to go and update both the `demo` code and the `example` code, which at the start differed from each other.  I think a better way forward would be to not have the demo directory or to somehow load the demo code directly into the example documentation.

**Comment by rstoneback** [comment]
Thanks @aburrell. I will start on this after class today.

File: demo/cnofs_vefi_dc_b_orbit_plots.py [hunk 1]
```diff
@@ -5,80 +5,74 @@
...
+    for i, pax in enumerate(ax):
+        if i == 0:
+            bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
+                vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
+            pax.bar(vefi['longitude'], vefi[p_params[i]] + 0.5,
+                    width=bwidth, bottom=-0.5)
+            pax.set_title(' - '.join((vefi.data.index[0].ctime(),
+                                      vefi.data.index[-1].ctime())))
+            pax.set_ylabel('Interp. Flag')
+            pax.set_ylim(-0.5, 1.5)
```
Comment (rstoneback) [inline demo/cnofs_vefi_dc_b_orbit_plots.py:60]:
```suggestion
            # Plot Interpolation status flag
            bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
                vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
            pax.plot(vefi['longitude'], vefi['B_flag'])
            pax.set_title(' - '.join((vefi.data.index[0].ctime(),
                                      vefi.data.index[-1].ctime())))
            pax.set_ylabel('Interp. Status')
            pax.set_ylim(0., 2.)
```

File: demo/cnofs_vefi_dc_b_orbit_plots.py [hunk 2]
```diff
@@ -5,80 +5,74 @@
...
 results_dir = ''
 
-# select vefi dc magnetometer data, use longitude to determine where
+# Select C/NOFS VEFI DC magnetometer data, use longitude to determine where
 # there are changes in the orbit (local time info not in file)
 orbit_info = {'index': 'longitude', 'kind': 'longitude'}
-vefi = pysat.Instrument(platform='cnofs', name='vefi', tag='dc_b',
-                        clean_level=None, orbit_info=orbit_info)
+vefi = pysat.Instrument(inst_module=pysatNASA.instruments.cnofs_vefi,
+                        tag='dc_b', clean_level=None, orbit_info=orbit_info)
```
Comment (rstoneback) [inline demo/cnofs_vefi_dc_b_orbit_plots.py:20]:
```suggestion
                        tag='dc_b', clean_level=None, orbit_info=orbit_info)
```

The change in plot comes from the clean_level setting. A setting of None uses the default `pysat.params['clean_level']`. To get no cleaning it needs to be set as `clean_level='none'` or we need to remove a setting in `pysat.params`

File: docs/examples/example_orbit.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                  vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
+              pax.bar(vefi['longitude'], vefi[p_params[i]] + 0.5,
+                      width=bwidth, bottom=-0.5)
+              pax.set_title(' - '.join((vefi.data.index[0].ctime(),
+                                        vefi.data.index[-1].ctime())))
+              pax.set_ylabel('Interp. Flag')
+              pax.set_ylim(-0.5, 1.5)
+          else:
+              pax.plot(vefi['longitude'], vefi[p_params[i]], '.', ms=1)
+              pax.set_title(vefi.meta[param].name)
```
Comment (rstoneback) [inline docs/examples/example_orbit.rst:73]:
```suggestion
              pax.set_title(vefi.meta[p_params[i]].name)
```

File: docs/examples/example_orbit.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+              pax.set_ylabel(vefi.meta[param].units)
```
Comment (rstoneback) [inline docs/examples/example_orbit.rst:74]:
```suggestion
              pax.set_ylabel(vefi.meta[p_params[i]].units)
```

File: docs/examples/example_orbit.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      for i, pax in enumerate(ax):
+          if i == 0:
+              bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
+                  vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
+              pax.bar(vefi['longitude'], vefi[p_params[i]] + 0.5,
+                      width=bwidth, bottom=-0.5)
+              pax.set_title(' - '.join((vefi.data.index[0].ctime(),
+                                        vefi.data.index[-1].ctime())))
+              pax.set_ylabel('Interp. Flag')
+              pax.set_ylim(-0.5, 1.5)
```
Comment (rstoneback) [inline docs/examples/example_orbit.rst:70]:
```suggestion
            # Plot Interpolation status flag
            bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
                vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
            pax.plot(vefi['longitude'], vefi['B_flag'])
            pax.set_title(' - '.join((vefi.data.index[0].ctime(),
                                      vefi.data.index[-1].ctime())))
            pax.set_ylabel('Interp. Status')
            pax.set_ylim(0., 2.)
```

File: docs/examples/example_orbit.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+   # Set the directory where the plots will be saved. Setting nothing will put
+   # the plots in the current directory
+   results_dir = ''
+
+   # Select C/NOFS VEFI DC magnetometer data, use longitude to determine where
+   # there are changes in the orbit (local time info not in file)
+   orbit_info = {'index': 'longitude', 'kind': 'longitude'}
+   vefi = pysat.Instrument(inst_module=pysatNASA.instruments.cnofs_vefi,
+	                   tag='dc_b', clean_level=None, orbit_info=orbit_info)
```
Comment (rstoneback) [inline docs/examples/example_orbit.rst:30]:
```suggestion
	                   tag='dc_b', clean_level='none', orbit_info=orbit_info)
```

**Comment by rstoneback** [comment]
We could make the demo code a Jupyter notebook and store it over in pysatDemo. From there we could print the code as a pdf and paste that in our docs.

I believe the original reason code was taken out of the docs and added to demo was because the cosmic comparison code had too many supporting functions. The docs kept the most important parts but the demo area had the full thing.

**Comment by rstoneback** [comment]
orbit_00000

File: demo/cnofs_vefi_dc_b_orbit_plots.py (hunk 1 above)
Comment (aburrell) [reply demo/cnofs_vefi_dc_b_orbit_plots.py:60]: So, you don't need `bwidth` without a bar.  I also like the bar plotting option for flags, just couldn't get any of the b_flag == 1 values.

File: demo/cnofs_vefi_dc_b_orbit_plots.py (hunk 2 above)
Comment (aburrell) [reply demo/cnofs_vefi_dc_b_orbit_plots.py:20]: We need to be very careful about this in the documentation, because that's a change in behaviour.

**Comment by aburrell** [comment]
Yeah, that was the original reason, but it never got carried through fully in the docs.  So, this is the pull request to do that!

**Comment by aburrell** [comment]
Ok, fixing `clean_level` makes the plot come out right.  I also found a few other errors that I fixed.  Now we just need to solve the problem of having the same code in `demo` and in the `examples` document.  If we want to kick the ball down the road, removing the demo directory and references and putting them in pysatDemo will do the trick for now.

File: demo/cnofs_vefi_dc_b_orbit_plots.py (hunk 1 above)
Comment (aburrell) [reply demo/cnofs_vefi_dc_b_orbit_plots.py:64]: As you can see from the new figure, one of the reasons I like the bar for this flag is that it makes it clearly visible when the interpolation flag doesn't have a value.

**Review by rstoneback — APPROVED** [review]
Thanks @aburrell. I'm ok with moving the code over to pysatDemo and kicking the can down the road to get this release out.

**Comment by aburrell** [comment]
@rstoneback I hereby task you with moving demo to pysatDemo so that I don't destroy it or send it into the æther.  Once that's done you or I can remove the demo dir in the PR and then convert this from a draft to a real, live PR.

**Comment by rstoneback** [comment]
pysatTutorials pull created: https://github.com/pysat/pysatTutorials/pull/3

**Comment by rstoneback** [comment]
> @rstoneback I hereby task you with moving demo to pysatDemo so that I don't destroy it or send it into the æther. Once that's done you or I can remove the demo dir in the PR and then convert this from a draft to a real, live PR.

@aburrell task complete.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**